### ISSUE TITLE
Default behavior to panic on status code != 200

### ISSUE THREAD

**Issue by morgenthum** [JosephLenton/axum-test#3]
I use axum-test to test a REST API. To me, it feels awkward to write
`let response = server.post("/sessions").expect_failure().json(&request).await;`
because I don't expect a failure - I expect a status code of 201 (Created), which signifies success in the context of a REST API. Perhaps it would be better to limit failure status codes to the 5xx range?

**Comment by NotNorom** [comment]
To add to that: I feel like `expect_failure()` and `expect_success()` aren't strictly necessary because of the various `assert_*` functions on `TestResponse` which are way more flexible.

**Comment by JosephLenton** [comment]
Hey @morgenthum, thanks for the feedback. Really expecting success and failure should be considering the 2xx range, and not only 200. I will be changing to this behaviour.

**Comment by JosephLenton** [comment]
@NotNorom I may add a flag to the config to opt out (or perhaps in?) to the fail fast behaviour.

The behaviour is something I explicitly wanted to have in this library based on my own years of writing E2E tests. As failing as quickly as possible in tests tends to lead to less debugging time.

**Comment by NotNorom** [comment]
@JosephLenton Fair enough! I agree the fail fast approach is nice :)

My main reason for my feedback was: I questioned myself what "failure" and "success" mean. For example I could have a test that requires a 403 return code considered to be a success. I could also see a 301 and 302 being considered as success.

Maybe renaming these functions would make sense? Obviously these are suggestions only:

1. `expect_1xx`, `expect_2xx`, `expect_3xx`, `expect_4xx`, `expect_5xx`
2. `expect_information`, `expect_success`, `expect_redirect`, `expect_client_error`, `expect_server_error`

Another suggestion, but this is probably overkill:
Have an argument for an `expect` function which allows combining the different states.
Something like

```rs
enum Response {
  Information,
  Success,
  Redirect,
  ClientError,
  ServerError,
}

let response = server
    .post(&"/login")
    .json(&json!({
        "user": "blah blah blah",
    }))
    .expect([Information, Success, Redirect])
    .await;

// Or maybe even implementing `std::ops::BitOr`:

let response = server
    .post(&"/login")
    .json(&json!({
        "user": "blah blah blah",
    }))
    .expect(Information | Success | Redirect)
    .await;
```

Just spit balling of course.

**Comment by JosephLenton** [comment]
This has been changed to matching on the whole 2xx range (or outside for failure) from this PR https://github.com/JosephLenton/axum-test/pull/11 , and released as version 9.0.0.
